modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2213,8 +2370,12 @@ def _compute_axis_labels_and_lengths(self, axis: int, partitions=None):
 
     def _build_index_cache(self):
         """Materialize index and store it in the cache."""
-        index, _ = self._compute_axis_labels_and_lengths(axis=0)
-        self.set_index_cache(index)
+        if self._partitions and not self._index_cols:
+            nrows = self._partitions[0][0]._length_cache
```

Comment:
is this guaranteed that we know lengths cache here? As far as I understand, a partition is obliged to know its length only when it's a 'raw' one, right?